architecture-decision-records/002-greenhouse-technical-implementation-of-access-authorization-for-plugins.md

```diff
@@ -0,0 +1,120 @@
+# 002 Technical Implementation of access Authorization for Greenhouse Plugins
+
+- Status: draft
+- Deciders: Fabian Ruff, Esther Schmitz, Arno Uhlig, Uwe Mayer, David Rochow
+- Date: 2023-03-09
+- Tags: greenhouse
+
+## Context and Problem Statement
+
+Greenhouse is a Platform that aims to aggregate a variety of Applications into a single Platform using a Plugin Concept
+that allows Applications to be integrated into Greenhouse while being maintained and developed in a distributed manner.
+
+Furthermore, it intentionally does not support multi-tenancy across Plugin Instances to enable segregation between
+tenants and make the Platform usable by totally unrelated LoB's.
+
+This Decision record is about the technical solution how we do Authorizations for Plugins.
+
+## Decision Drivers
+
+- Enables support of multiple Identity Providers
+    * To allow Organizations to use their own IdP
+- Open for adoption
+    * allows also Plugin Backends to be used that are not developed internally
+- Support of Role Mapping within Greenhouse
+    * Supports the usage for the considered solutions of ADR-1
+- Supports running Greenhouse components in a single Kubernetes Cluster
+    * On kubernetes you can only configure one OIDC Provider
+- Implementation effort
+- Maintenance effort
+
+## Considered Options
+
+- Team Sync during Deployment
+- Team Sync by Plugin during runtime
+- Usage of internal IdP for Group Ownership Rewriting based on Greenhouse mappings
+
+## Decision Outcome
+
+Chosen option: **"Usage of internal IdP for Group Ownership Rewriting based on Greenhouse mappings"**
+
+### Positive Consequences
+
+- Overall best decision driver ratings
+- Most flexible solution
+- Does not require additional syncing of mappings between Greenhouse and Plugins
+- We are in control of the OIDC Provider that is used for Authorization of Requests on Plugins
+- The authentication is still happening on the external Identity Provider
+- Only of the Solutions that solves the Kubernetes problem(just supports one OIDC Provider) by design
+
+### Negative Consequences
+
+- Introduction of an additional Open Source Project
+- In case we need to support Plugin Backends outside Converged Cloud, we would need to expose the internal OIDC
+  Provider (somehow) or build an additional proxy solution.
+- This solution is expected to require the most implementation and maintenance effort
+
+## Pros and Cons of the Options
+
+### Team Sync during Deployment
+
+![](./assets/a0b55e95-8ded-47bb-96ce-67729b3f123c.png)
```

Comment:
This does not resolve properly when running
`log4brains preview`
as it tries to serve the image from `./assets`
Not sure we can solve this and keep the image in the markdown intact?

Reply:
yes, one of the drawbacks...

In a deployed version we would need to copy the assets into  `.log4brains/out/l4b-static/` and in the markdown reference it as `![my-image](/l4b-static/images/my-image.png)`